### PR TITLE
[BUG] Wrong breadcrumbs after browser reload on edit detectors page #395

### LINKED ISSUE #395: [BUG] Wrong breadcrumbs after browser reload on edit detectors page
**What is the bug?**
Wrong breadcrumbs after browser reload on edit detectors page.

**How can one reproduce the bug?**
1. Go to the detectors page
2. Click on any detector to open the detector details page
3. Edit detector details
4. When on the detector details page reload the page
5. Notice the breadcrumbs showing the wrong path

**What is the expected behavior?**
There should be a correct path in the breadcrumbs.

**Do you have any screenshots?**
Page with bad breadcrumbs path.
edit details wrong breadcrumbs
Page with good breadcrumbs path.
edit details correct breadcrumbs

### PR DESCRIPTION
**Pull request by jovancacvetkovic**
Signed-off-by: Jovan Cvetkovic <jovanca.cvetkovic@gmail.com>

### Description
Fixes broken pages after the browser reload.

### Issues Resolved
Resolves #395 

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by amsiglan — APPROVED** [review]

File: public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx [hunk 1]
```diff
@@ -74,8 +84,24 @@ export default class UpdateAlertConditions extends Component<
...
+      const { ruleService, detectorService } = this.props;
       const { detector } = this.state;
+      if (!detector.name) {
+        const response = (await detectorService.getDetectors()) as ServerResponse<
+          SearchDetectorsResponse
+        >;
+        if (response.ok) {
+          const detectorHit = response.response.hits.hits.find(
+            (detectorHit) => detectorHit._id === detector.id
+          ) as DetectorHit;
```
Comment (amsiglan) [inline public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:96]: We should instead use the get detector API using the id - https://opensearch.org/docs/latest/security-analytics/api-tools/detector-api/#get-detector

**Review by amsiglan — CHANGES_REQUESTED** [review]
Requesting change in API as mentioned in another comment

File: public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx (hunk 1 above)
Comment (jovancacvetkovic) [reply public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:96]: @amsiglan I noticed we are using this method in other places as well and as I can see, the reason behind it is that the get detector doesn't return the triggers array in its response, and we need this for alerts. Perhaps, we could add a feature request for BE to add triggers in response as well?

Comment (amsiglan) [reply public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:96]: aah, let me add an issue, but most likely it won't be available before 2.6, so we can go ahead with current approach for now

**Review by amsiglan — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**